File: 0686-repeated-string-match/0686-repeated-string-match.py
class Solution(object):
    def repeatedStringMatch(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: int
        """
        # times = -(-len(B) // len(A))  # Equal to ceil(len(B) / len(A))
        # print(times)
        # for i in range(2):
        #     print((A * (times + i)))
        #     if B in (A * (times + i)):
        #         return times + i
        # return -1
    
    
    
        my_str = a
        count = 1
        n = len(b)//len(a)
        
        for i in range(n+2) :
            if b in (my_str*(i+1)) :
                return count
            else :
                count += 1
                # my_str += a
        
        return -1

File: 0686-repeated-string-match/0686-repeated-string-match.cpp
class Solution {
public:
    int repeatedStringMatch(string a, string b) {
       
       string str=a;
       int count=1;
       int repeat=b.length()/a.length();
       for(int i=0; i<(repeat+2); i++){
           if(str.find(b) != string::npos){
               return count;
           }else{
               str+=a;
               count++;
           }
       }
       return -1;
    }
};

